invokeai/backend/tiles/tiles.py

```diff
@@ -63,31 +97,122 @@ def calc_tiles_with_overlap(
+            left = tile_idx_x * (tile_size_x - overlap_x)
+            right = min(left + tile_size_x, image_width)
+            # For the last tile in the row adjust right to be the width of the image
+            if tile_idx_x == num_tiles_x - 1:
+                right = image_width
 
-            assert cur_tile is not None
+            tile = Tile(
+                coords=TBLR(top=top, bottom=bottom, left=left, right=right),
+                overlap=TBLR(top=0, bottom=0, left=0, right=0),
+            )
 
-            # Update cur_tile top-overlap and corresponding top-neighbor bottom-overlap.
-            if top_neighbor_tile is not None:
-                cur_tile.overlap.top = max(0, top_neighbor_tile.coords.bottom - cur_tile.coords.top)
-                top_neighbor_tile.overlap.bottom = cur_tile.overlap.top
+            tiles.append(tile)
 
-            # Update cur_tile left-overlap and corresponding left-neighbor right-overlap.
-            if left_neighbor_tile is not None:
-                cur_tile.overlap.left = max(0, left_neighbor_tile.coords.right - cur_tile.coords.left)
-                left_neighbor_tile.overlap.right = cur_tile.overlap.left
+    return calc_overlap(tiles, num_tiles_x, num_tiles_y)
 
-    return tiles
+
+def calc_tiles_min_overlap(
+    image_height: int,
+    image_width: int,
+    tile_height: int,
+    tile_width: int,
+    min_overlap: int = 0,
+    round_to_8: bool = False,
+) -> list[Tile]:
+    """Calculate the tile coordinates for a given image shape under a simple tiling scheme with overlaps.
+
+    Args:
+        image_height (int): The image height in px.
+        image_width (int): The image width in px.
+        tile_height (int): The tile height in px. All tiles will have this height.
+        tile_width (int): The tile width in px. All tiles will have this width.
+        min_overlap (int): The target minimum overlap between adjacent tiles. If the tiles do not evenly cover the image
+            shape, then the overlap will be spread between the tiles.
+
+    Returns:
+        list[Tile]: A list of tiles that cover the image shape. Ordered from left-to-right, top-to-bottom.
+    """
+    assert image_height >= tile_height
+    assert image_width >= tile_width
+    assert min_overlap < tile_height
+    assert min_overlap < tile_width
+
+    num_tiles_x = math.ceil((image_width - min_overlap) / (tile_width - min_overlap)) if tile_width < image_width else 1
+    num_tiles_y = (
+        math.ceil((image_height - min_overlap) / (tile_height - min_overlap)) if tile_height < image_height else 1
+    )
+
+    # tiles[y * num_tiles_x + x] is the tile for the y'th row, x'th column.
+    tiles: list[Tile] = []
+
+    # Calculate tile coordinates. (Ignore overlap values for now.)
+    for tile_idx_y in range(num_tiles_y):
+        top = (tile_idx_y * (image_height - tile_height)) // (num_tiles_y - 1) if num_tiles_y > 1 else 0
+        if round_to_8:
+            top = 8 * (top // 8)
```

Comment:
This is also @JPPhoto code so hopefully, he can comment on it.

Reply:
@RyanJDick The initial reason we were rounding top to 8 was to support larger noise fields if the user wanted that - basically think of a noise field as big as the final latent. If we're doing that, we need to round coordinates to 8 to be compatible with latent sizes.

That said, I didn't personally think it was that useful of a feature and I'm in favor of dropping the `if round_to_8` conditions entirely. @skunkworxdark Thoughts?

I do think it's fine to assert that tile_width and tile_height are multiples of 8, or to enforce that with typing.

Reply:
I don't think cropping from a large noise fields had a definitively positive impact.  When it did produce better results it could easily be down to luck of what noise was chosen just in the same way having a good random seed for noise currently is.

However, I don't think it does any harm in keeping it in as an option either as it doesn't seem to operate any worse that just having a flat unchanging noise for each tile. My feeling is that we have not done enough testing to be definitive either way.  I could easily be that some combination of schedulers or options might have better results with it.  I still feel that it has validty as an option.

Reply:
I think in that case my gut tells me to yank it as it adds unnecessary complexity, and we can add it back in at a higher level down the road - maybe not the splitter but the client of `CalculateImageTilesOutput` should be the responsible party for doing the rounding work.

Reply:
@JPPhoto This particular routine `calc_tiles_min_overlap` is mostly your code so I am OK going in the direction the majority wants.   I suppose if a user wants to do tile noise pulled from an output image seized latent noise then they can always just calculate the exact tile and overlap sizes that don't break it or use the even_split version :)